src/components/PeoplePage.tsx

```diff
@@ -1,35 +1,201 @@
+        throw error;
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const peopleFromServer = people.map(person => ({
+    ...person,
+    mother: people.find(p => p.name === person.motherName),
+    father: people.find(p => p.name === person.fatherName),
+    century: String(Number(person.born.toString().slice(0, 2)) + 1),
+  }));
+
+  const { slug } = useParams();
+
+  const normalizedQuery = query.toLowerCase().trim();
+
+  const isIncludesNormalizedQuery = (name: string) => {
+    return name.toLowerCase().includes(normalizedQuery)
+  }
+
+  let filteredPeople = peopleFromServer.filter(p => {
+    let personName = isIncludesNormalizedQuery(p.name);
+    let personFatherName =  isIncludesNormalizedQuery(p.fatherName ?? '');
+    let personMotherName = isIncludesNormalizedQuery(p.motherName ?? '');
+
+    return personName || personFatherName || personMotherName
+  });
+
+  if (sex) {
+    filteredPeople = filteredPeople.filter(p => p.sex === sex);
+  }
+
+  if (centuries.length > 0) {
+    filteredPeople = filteredPeople.filter(p => (
+      centuries.includes(p.century)
+    ));
+  }
+
+  switch (sort) {
+    case SortTypes.name:
+      filteredPeople.sort((value1, value2) => {
+        return value1.name.localeCompare(value2.name);
+      });
+      break;
+
+    case SortTypes.sex:
+      filteredPeople.sort((value1, value2) => {
+        return value1.sex.localeCompare(value2.sex);
+      });
+      break;
+
+    case SortTypes.born:
+      filteredPeople.sort((value1, value2) => {
+        return value1.born - value2.born;
+      });
+      break;
+
+    case SortTypes.died:
+      filteredPeople.sort((value1, value2) => {
+        return value1.born - value2.born;
+      });
+      break;
+  }
+
+  switch (order) {
+    case 'desc':
+      filteredPeople = filteredPeople.reverse();
+      break;
+  }
+
+  const allCategories = ['Name', 'Sex', 'Born', 'Died'];
+
   return (
     <>
       <h1 className="title">People Page</h1>
 
-      <div className="block">
-        <div className="columns is-desktop is-flex-direction-row-reverse">
-          <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
-          </div>
-
-          <div className="column">
-            <div className="box table-container">
-              <Loader />
+      {people.length !== 0 ? (
+        <div className="block">
+          <div className="columns is-desktop is-flex-direction-row-reverse">
+            <div className="column is-7-tablet is-narrow-desktop">
+              <PeopleFilters
+                setSearchParams={setSearchParams}
+                searchParams={searchParams}
+                centuries={centuries}
+                query={query}
+                sex={sex}
+              />
+            </div>
 
-              <p data-cy="peopleLoadingError">Something went wrong</p>
+            <div className="column">
+              <div className="box table-container">
+                <table
```

Comment:
Table should not be seen when there is no filtered people. You should show message instead of table